yt_dlp/extractor/picarto.py

```diff
@@ -102,21 +111,30 @@ class PicartoVodIE(InfoExtractor):
     def _real_extract(self, url):
         video_id = self._match_id(url)
 
-        webpage = self._download_webpage(url, video_id)
+        data = self._download_json(
+            'https://ptvintern.picarto.tv/ptvapi', video_id, query={
```

Comment:
This shares some common code with the API call in `PicartoIE._real_extract`, but not enough to make adding a baseclass worth it IMO